crates/ink/codegen/src/generator/mod.rs

```diff
@@ -70,3 +73,32 @@ pub use self::{
     storage_item::StorageItem,
     trait_def::TraitDefinition,
 };
+
+/// Check if the the conditional compilation expression
+/// for any `cfg` attributes is satisfied.
+///
+/// # Example
+/// If we have an annotated item (e.g. message)
+/// ```ignore
+/// #[ink(message)]
+/// #[cfg(feature = "foo")]
+/// #[cfg(feature = "baz")]
+/// pub fn get(&self) -> String {
+///     self.value.clone()
+/// }
+/// ```
+/// The function would would iterate over the `cfg` attributes
+/// and return `true` if any of the feature flags are set
+pub fn is_conditionally_excluded(attrs: &[Attribute]) -> bool {
+    for attr_tokens in attrs
+        .iter()
+        .filter(|a| a.path.is_ident("cfg"))
+        .map(|a| &a.tokens)
+    {
+        let _s = attr_tokens.to_token_stream();
+        if !cfg!(_s) {
```

Comment:
Not 100% sure of the logic here I will check back tomorrow

Reply:
I extract tokens from the `cfg` attribute and pass them into `cfg!` macro which allows to filter our code spans

Reply:
Ok I think the logic is correct but it is at odds with the comments